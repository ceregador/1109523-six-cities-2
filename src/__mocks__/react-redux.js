export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
