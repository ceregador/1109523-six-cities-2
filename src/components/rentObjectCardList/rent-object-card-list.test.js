import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import Renderer from 'react-test-renderer';
import RentObjectCardList from '../rentObjectCardList/rent-object-card-list.jsx';
import RentObjectCard from '../rentObjectCard/rent-object-card.jsx';
import Sorting from '../sorting/sorting.jsx';

jest.mock(`../rentObjectCard/rent-object-card.jsx`);
jest.mock(`../sorting/sorting.jsx`);
jest.mock(`react-redux`);

it(`renders and connnects correctly`, () => {
  const tree = Renderer
    .create(
        <RentObjectCardList
          cityName={``}
          offers={[
            {
              id: 0,
              city: {name: ``},
              name: ``,
              isPremium: false,
              image: ``,
              type: ``,
              price: 1,
              isBookmarked: false,
              onTitleClick: () => null,
              onActiveOfferChanged: () => null
            }
          ]}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), null);

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({offers: []});
  expect(mappedProps).toHaveProperty(`offers`);

  expect(connectAdvanced).toHaveBeenCalledWith(RentObjectCardList);

  expect(RentObjectCard).toHaveBeenCalled();
  expect(Sorting).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
