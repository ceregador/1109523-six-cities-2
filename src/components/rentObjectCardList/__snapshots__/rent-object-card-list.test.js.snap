// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and connnects correctly 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    1
     places to stay in 
    
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <select
      className="places__sorting-type"
      id="places-sorting"
    >
      <option
        className="places__option"
        selected=""
        value="popular"
      >
        Popular
      </option>
      <option
        className="places__option"
        value="to-high"
      >
        Price: low to high
      </option>
      <option
        className="places__option"
        value="to-low"
      >
        Price: high to low
      </option>
      <option
        className="places__option"
        value="top-rated"
      >
        Top rated first
      </option>
    </select>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  />
</section>
`;
